client/src/components/SignUpForm.js

```diff
@@ -37,6 +47,19 @@ export default function SignUp() {
         }
         setOpen(false);
     };
+
+    const handleChange = (e) => {
+        setAddress(e.target.value);
+        address && address.length > 3 &&
```

Comment:
^ That would throw an error if address is undefined/null, so if it's possible for that to happen we should keep the `address &&`

Reply:
^ That would throw an error if address is undefined/null, so if it's possible for that to happen we should keep the `address &&`

Reply:
I don't think it can happen because he sets it to an empty string initially. Could still be null if the user can provides a null value or one of predictions is null.